packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionHistory.tsx

```diff
@@ -1,308 +1,128 @@
+    transactions,
+    address,
+    loading,
+    completed,
+    error,
+    failedChainPairs,
+    resume
+  } = props
+
+  const oldestTxTimeAgoString = useMemo(() => {
+    return dayjs(transactions[transactions.length - 1]?.createdAt).toNow(true)
+  }, [transactions])
+
+  const groupedTransactions = useMemo(
+    () =>
+      transactions.reduce(
+        (acc, tx) => {
+          if (isTxCompleted(tx) || isTxExpired(tx)) {
+            acc.settled.push(tx)
+          }
+          if (isTxPending(tx)) {
+            acc.pending.push(tx)
+          }
+          if (isTxClaimable(tx)) {
+            acc.claimable.push(tx)
+          }
+          if (isTxFailed(tx)) {
+            acc.failed.push(tx)
+          }
+          return acc
+        },
+        {
+          settled: [] as MergedTransaction[],
+          pending: [] as MergedTransaction[],
+          claimable: [] as MergedTransaction[],
+          failed: [] as MergedTransaction[]
         }
-
-        if (t2.blockNum && t1.blockNum) {
-          return t2.blockNum - t1.blockNum
-        }
-
-        return 0
-      })
-    return transactions.filter(tx => isPending(tx))
-  }, [mergedTransactions, pendingIdsCctp, transfersCctp])
-
-  const failedTransactions = useMemo(() => {
-    return mergedTransactions.filter(tx => isFailed(tx))
-  }, [mergedTransactions])
-
-  const roundedTabClasses =
-    'roundedTab ui-not-selected:arb-hover relative flex flex-row flex-nowrap items-center gap-0.5 md:gap-2 rounded-tl-lg rounded-tr-lg px-2 md:px-4 py-2 text-base ui-selected:bg-white ui-not-selected:text-white justify-center md:justify-start grow md:grow-0'
-
-  const handleSmartContractWalletTxHistoryTab = useCallback(
-    (index: number) => {
-      if (!isSmartContractWallet || !chain) {
-        return
-      }
-      const isCctpTab = index === TransactionHistoryTab.CCTP
-      const isConnectedToArbitrum = isNetwork(chain.id).isArbitrum
-
-      if (isCctpTab) {
-        if (isConnectedToArbitrum) {
-          showCctpDepositsTransactions()
-        } else {
-          showCctpWithdrawalsTransactions()
-        }
-      }
-    },
-    [
-      chain,
-      isSmartContractWallet,
-      showCctpDepositsTransactions,
-      showCctpWithdrawalsTransactions
-    ]
+      ),
+    [transactions]
   )
 
-  useEffect(() => {
-    // this function runs every time the network tab is changed, and here it is also triggered when the page loads
-    // it sets the tab to 0 (deposits), which is the default tab
-    handleSmartContractWalletTxHistoryTab(0)
-  }, [handleSmartContractWalletTxHistoryTab])
-
-  useEffect(() => {
-    // This check avoid the situation when we open the transaction history on CCTP tab after a transfer, but it would go back to "To Arbitrum" tab
-    if (transfersIds.length === 0) {
-      setTransactionHistoryTab(0)
-    }
-  }, [address, chain, setTransactionHistoryTab, transfersIds])
+  const pendingTransactions = [
```

Comment:
Q: So we aren't sorting the pending transactions by time, just grouping by failed-pending-claimable?

Reply:
They are already sorted by time when returned from the hook. Here we just group them so specific types appear on top